File: app/src/main/java/net/nashihara/naroureader/utils/RealmUtils.java
package net.nashihara.naroureader.utils;

import android.content.Context;

import io.realm.Realm;
import io.realm.RealmConfiguration;

public class RealmUtils {
    private static Migration migration = new Migration();
    private static RealmConfiguration defaultConfig;

    private static int VERSION = 3;

    public static Realm getRealm(Context context) {
        if (defaultConfig == null) {
            Realm.init(context);
            defaultConfig = getConfig();
        }

        return Realm.getInstance(defaultConfig);
    }

    private static RealmConfiguration getConfig() {
        defaultConfig = new RealmConfiguration.Builder()
            .schemaVersion(VERSION)
            .migration(migration)
            .build();
        return defaultConfig;
    }

}

File: app/src/main/java/net/nashihara/naroureader/utils/RealmUtils.kt
package net.nashihara.naroureader.utils

import android.content.Context

import io.realm.Realm
import io.realm.RealmConfiguration
import okhttp3.internal.io.RealConnection

object RealmUtils {
    private val migration = Migration()
    private var defaultConfig: RealmConfiguration? = null

    private val VERSION = 3

    fun getRealm(context: Context): Realm {
        if (defaultConfig == null) {
            Realm.init(context)
            defaultConfig = buildConfig()
        }

        return Realm.getInstance(defaultConfig)
    }

    private fun buildConfig() : RealmConfiguration? {
        return RealmConfiguration.Builder()
                .schemaVersion(VERSION.toLong())
                .migration(migration)
                .build()
    }
}
